### PR TITLE
Keybindings config

### PR DESCRIPTION
**Pull request by elferherrera**
Adds the ability to parse simple keybindings into the emacs editor

To add an event you need to add this to your config file
```
 keybindings: [
        {
          name: completion
           modifier: CONTROL
           keycode: Char_c
           mode: emacs
           event: {
             type: single
             event: Complete
           }
         }
         {
          name: menu                                                                                                                                                                                  modifier: CONTROL
           keycode: down
           mode: emacs
           event: {
             type: single
             event: ContextMenu
           }
         }
       ]
```

Note: There is a refactoring of main function to locate utility functions

### CODE REVIEW

File: src/fuzzy_completion.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+use dialoguer::{
```
Comment (fdncred) [inline src/fuzzy_completion.rs:1]: #766 just removed all this stuff I think. is there a reason we need it?

File: src/reedline_config.rs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            ReedlineEvent::Edit(vec![EditCommand::InsertChar('p')]),
+            ReedlineEvent::Enter,
+        ]),
+    );
+
+    for keybinding in parsed_keybindings {
+        let modifier = match keybinding.modifier.as_str() {
+            "CONTROL" => KeyModifiers::CONTROL,
+            "SHIFT" => KeyModifiers::CONTROL,
+            _ => unimplemented!(),
```
Comment (fdncred) [inline src/reedline_config.rs:79]: why only control and shift and not option/alt?

File: src/reedline_config.rs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        ReedlineEvent::Multiple(vec![
+            ReedlineEvent::Edit(vec![EditCommand::InsertChar('p')]),
+            ReedlineEvent::Enter,
+        ]),
+    );
+
+    for keybinding in parsed_keybindings {
+        let modifier = match keybinding.modifier.as_str() {
+            "CONTROL" => KeyModifiers::CONTROL,
+            "SHIFT" => KeyModifiers::CONTROL,
```
Comment (fdncred) [inline src/reedline_config.rs:78]: shouldn't this be `KeyModifiers::SHIFT`?

**Comment by jmoore34** [comment]
Looks amazing with the record syntax, 10/10

File: src/fuzzy_completion.rs (hunk 1 above)
Comment (elferherrera) [reply src/fuzzy_completion.rs:1]: None. I just didn't see the merge. I'll delete it

File: src/reedline_config.rs (hunk 2 above)
Comment (elferherrera) [reply src/reedline_config.rs:79]: Lazynes. I want to merge this before continuing

File: src/reedline_config.rs (hunk 3 above)
Comment (elferherrera) [reply src/reedline_config.rs:78]: You are right. I'll continue working on this

**Pull request merged**